# Analysis of the California Regional Multiple Listing Service (CRMLS) Data

This will explore datasets related to Califlornia real estate data from January 2024 - April 2026

<hr>

by: YuLi Lin

### **Week 1** — Monthly Dataset Aggregation

#### All imports

In [356]:
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns; sns.set()
import numpy as np

##### Aggregating all listing files into "list"

In [357]:
list1 = pd.read_csv("CRMLSListing202401.csv")
list2 = pd.read_csv("CRMLSListing202402.csv")
list3 = pd.read_csv("CRMLSListing202403.csv")
list4 = pd.read_csv("CRMLSListing202404.csv")
list5 = pd.read_csv("CRMLSListing202405.csv")
list6 = pd.read_csv("CRMLSListing202406.csv")
list7 = pd.read_csv("CRMLSListing202407.csv")
list8 = pd.read_csv("CRMLSListing202408.csv")
list9 = pd.read_csv("CRMLSListing202409.csv")
list10 = pd.read_csv("CRMLSListing202410.csv")
list11 = pd.read_csv("CRMLSListing202411.csv")
list12 = pd.read_csv("CRMLSListing202412.csv")
list13 = pd.read_csv("CRMLSListing202501.csv")
list14 = pd.read_csv("CRMLSListing202502.csv")
list15 = pd.read_csv("CRMLSListing202503.csv")
list16 = pd.read_csv("CRMLSListing202504.csv")
list17 = pd.read_csv("CRMLSListing202505.csv")
list18 = pd.read_csv("CRMLSListing202506.csv")
list19 = pd.read_csv("CRMLSListing202507.csv")
list20 = pd.read_csv("CRMLSListing202508.csv")
list21 = pd.read_csv("CRMLSListing202509.csv")
list22 = pd.read_csv("CRMLSListing202510.csv")
list23 = pd.read_csv("CRMLSListing202511.csv")
list24 = pd.read_csv("CRMLSListing202512.csv")
list25 = pd.read_csv("CRMLSListing202601.csv")
list26 = pd.read_csv("CRMLSListing202602.csv")
list27 = pd.read_csv("CRMLSListing202603.csv")
list28 = pd.read_csv("CRMLSListing202604.csv")

list = pd.concat([globals()[f'list{i}'] for i in range(1, 29)], ignore_index=True)

C:\Users\liny0\AppData\Local\Temp\ipykernel_26144\199253818.py:26: DtypeWarning: Columns (32,58,63,66) have mixed types. Specify dtype option on import or set low_memory=False.
  list26 = pd.read_csv("CRMLSListing202602.csv")


##### Confirming row count before and after concatenation for listing data

In [358]:
rows_list_pre = sum(len(globals()[f'list{i}']) for i in range(1, 29))

rows_list_post = len(list)

print(f"Before: {rows_list_pre:,} | After: {rows_list_post:,}")

Before: 893,594 | After: 893,594


##### Aggregating all sold files into "sold"

In [359]:
sold1 = pd.read_csv("CRMLSSold202401_filled.csv").drop(columns = ["latfilled", "lonfilled"])
sold2 = pd.read_csv("CRMLSSold202401.csv")
sold3 = pd.read_csv("CRMLSSold202402.csv")
sold4 = pd.read_csv("CRMLSSold202403_filled.csv").drop(columns = ["latfilled", "lonfilled"])
sold5 = pd.read_csv("CRMLSSold202403.csv")
sold6 = pd.read_csv("CRMLSSold202404_filled.csv").drop(columns = ["latfilled", "lonfilled"])
sold7 = pd.read_csv("CRMLSSold202404.csv")
sold8 = pd.read_csv("CRMLSSold202405_filled.csv").drop(columns = ["latfilled", "lonfilled"])
sold9 = pd.read_csv("CRMLSSold202406_filled.csv").drop(columns = ["latfilled", "lonfilled"])
sold10 = pd.read_csv("CRMLSSold202407_filled.csv").drop(columns = ["latfilled", "lonfilled"])
sold11 = pd.read_csv("CRMLSSold202408.csv")
sold12 = pd.read_csv("CRMLSSold202409.csv")
sold13 = pd.read_csv("CRMLSSold202410.csv")
sold14 = pd.read_csv("CRMLSSold202411.csv")
sold15 = pd.read_csv("CRMLSSold202412.csv")
sold16 = pd.read_csv("CRMLSSold202501_filled.csv").drop(columns = ["latfilled", "lonfilled"])
sold17 = pd.read_csv("CRMLSSold202502.csv")
sold18 = pd.read_csv("CRMLSSold202503.csv")
sold19 = pd.read_csv("CRMLSSold202504.csv")
sold20 = pd.read_csv("CRMLSSold202505.csv")
sold21 = pd.read_csv("CRMLSSold202506.csv")
sold22 = pd.read_csv("CRMLSSold202507.csv")
sold23 = pd.read_csv("CRMLSSold202508.csv")
sold24 = pd.read_csv("CRMLSSold202509.csv")
sold25 = pd.read_csv("CRMLSSold202510.csv")
sold26 = pd.read_csv("CRMLSSold202511.csv")
sold27 = pd.read_csv("CRMLSSold202512.csv")
sold28 = pd.read_csv("CRMLSSold202601.csv")
sold29 = pd.read_csv("CRMLSSold202602.csv")
sold30 = pd.read_csv("CRMLSSold202603.csv")
sold31 = pd.read_csv("CRMLSSold202604.csv")

sold = pd.concat([globals()[f'sold{i}'] for i in range(1, 32)], ignore_index=True)

C:\Users\liny0\AppData\Local\Temp\ipykernel_26144\1936171636.py:6: DtypeWarning: Columns (2,36,39,56,74) have mixed types. Specify dtype option on import or set low_memory=False.
  sold6 = pd.read_csv("CRMLSSold202404_filled.csv").drop(columns = ["latfilled", "lonfilled"])
C:\Users\liny0\AppData\Local\Temp\ipykernel_26144\1936171636.py:7: DtypeWarning: Columns (2,36,39,56,74) have mixed types. Specify dtype option on import or set low_memory=False.
  sold7 = pd.read_csv("CRMLSSold202404.csv")
C:\Users\liny0\AppData\Local\Temp\ipykernel_26144\1936171636.py:21: DtypeWarning: Columns (4) have mixed types. Specify dtype option on import or set low_memory=False.
  sold21 = pd.read_csv("CRMLSSold202506.csv")
C:\Users\liny0\AppData\Local\Temp\ipykernel_26144\1936171636.py:28: DtypeWarning: Columns (4,74) have mixed types. Specify dtype option on import or set low_memory=False.
  sold28 = pd.read_csv("CRMLSSold202601.csv")


##### Confirming row count before and after concatenation for sold data

In [360]:
rows_sold_pre = sum(len(globals()[f'sold{i}']) for i in range(1, 32))

rows_sold_post = len(sold)

print(f"Before: {rows_sold_pre:,} | After: {rows_sold_post:,}")

Before: 681,599 | After: 681,599


##### Filtering "list" and "sold" to PropertyType == "Residential" and confirming row count before and after Residential filter

In [361]:
list_resid = list[list["PropertyType"] == "Residential"]
print(f"(Listing) Before: {len(list):,} | After: {len(list_resid):,}")

(Listing) Before: 893,594 | After: 567,549


In [362]:
sold_resid = sold[sold["PropertyType"] == "Residential"]
print(f"(Sold) Before:{len(sold):,} | After: {len(sold_resid):,}")

(Sold) Before:681,599 | After: 458,336


#### Save as new CSV files (w/ residential only filter)

In [363]:
list_resid.to_csv("w1_list_resid.csv", index=False)
sold_resid.to_csv("w1_sold_resid.csv", index=False)

<hr>

### **Week 2** — Dataset Structuring and Validation (EDA)

#### Inspecting structure

In [ ]:
sold.columns

In [365]:
pd.set_option('display.max_columns', None)
sold.head()

,Flooring,ViewYN,WaterfrontYN,BasementYN,PoolPrivateYN,OriginalListPrice,ListingKey,ListAgentEmail,CloseDate,ClosePrice,ListAgentFirstName,ListAgentLastName,Latitude,Longitude,UnparsedAddress,PropertyType,LivingArea,ListPrice,DaysOnMarket,ListOfficeName,BuyerOfficeName,CoListOfficeName,ListAgentFullName,CoListAgentFirstName,CoListAgentLastName,BuyerAgentMlsId,BuyerAgentFirstName,BuyerAgentLastName,FireplacesTotal,AssociationFeeFrequency,AboveGradeFinishedArea,ListingKeyNumeric,MLSAreaMajor,TaxAnnualAmount,CountyOrParish,MlsStatus,ElementarySchool,AttachedGarageYN,ParkingTotal,BuilderName,PropertySubType,LotSizeAcres,SubdivisionName,BuyerOfficeAOR,YearBuilt,BuyerAgencyCompensationType,StreetNumberNumeric,ListingId,BathroomsTotalInteger,City,BuyerAgencyCompensation,TaxYear,BuildingAreaTotal,BedroomsTotal,ContractStatusChangeDate,ElementarySchoolDistrict,CoBuyerAgentFirstName,PurchaseContractDate,ListingContractDate,BelowGradeFinishedArea,BusinessType,StateOrProvince,CoveredSpaces,MiddleOrJuniorSchool,FireplaceYN,Stories,HighSchool,Levels,LotSizeDimensions,LotSizeArea,MainLevelBedrooms,NewConstructionYN,GarageSpaces,HighSchoolDistrict,PostalCode,AssociationFee,LotSizeSquareFeet,MiddleOrJuniorSchoolDistrict,BuyerAgentAOR,ListAgentAOR,OriginatingSystemName,OriginatingSystemSubName
0,"Carpet,Tile,Wood",True,NaN,NaN,False,499000.0,551985747,jwachter@cbnorcal.com,2024-01-26,240000.0,Joan,Wachter,37.566106,-122.327954,1 Baldwin Avenue 411,Residential,1140.0,295000.0,777,Coldwell Banker Realty,Bay Area Senior Services,NaN,Joan Wachter,NaN,NaN,ML5090968,Scott,Withrow,NaN,Monthly,NaN,551985747,699 - Not Defined,NaN,San Mateo,Closed,NaN,False,1.0,NaN,Condominium,NaN,NaN,MLSListings,1988.0,Item1,1.0,ML81865679,2.0,San Mateo,2.5,NaN,NaN,2.0,2024-01-26,NaN,NaN,2023-11-22,2021-10-06,NaN,NaN,CA,NaN,NaN,True,NaN,NaN,NaN,NaN,NaN,NaN,False,1.0,Other,94401,6472.0,NaN,NaN,NaN,NaN,NaN,NaN
1,NaN,NaN,NaN,NaN,NaN,0.0,535486633,eabrown@lee-associates.com,2024-01-24,950.0,Elizabeth,Brown,34.529407,-117.319674,15770 Mojave Drive L,CommercialLease,NaN,808.0,901,Lee & Associates,Lee & Associates,NaN,Elizabeth Brown,NaN,NaN,HD11042,Elizabeth,Brown,NaN,NaN,NaN,535486633,NaN,NaN,San Bernardino,Closed,NaN,NaN,NaN,NaN,Retail,1.2000,NaN,HighDesert,1980.0,Item1,15770.0,538013,NaN,Victorville,3.0,NaN,NaN,NaN,2024-01-24,NaN,NaN,2024-01-24,2021-08-06,NaN,Other,CA,NaN,NaN,NaN,NaN,NaN,NaN,NaN,52320.0,NaN,NaN,NaN,NaN,92394,NaN,52320.0,NaN,NaN,NaN,NaN,NaN
2,NaN,True,NaN,NaN,NaN,75000.0,529986282,Joe@9WINWIN.com,2024-01-16,45000.0,Joseph,Piteleski,35.618009,-118.473141,0 Shadow mountain Dr.,Land,NaN,75000.0,865,Keller Williams OC Coastal Realty,Keller Williams OC Coastal Realty,NaN,Joseph Piteleski,NaN,NaN,spitejos,Joseph,Piteleski,NaN,NaN,NaN,529986282,LKIS - Lake Isabella,NaN,Kern,Closed,NaN,NaN,NaN,NaN,NaN,4.9900,NaN,OrangeCounty,NaN,Item1,0.0,OC21157144,NaN,Lake Isabella,3.0,NaN,NaN,NaN,2024-01-16,NaN,NaN,2023-12-22,2021-07-18,NaN,NaN,CA,NaN,NaN,NaN,NaN,NaN,NaN,NaN,217364.0,NaN,False,NaN,NaN,93240,NaN,217364.0,NaN,NaN,NaN,NaN,NaN
3,NaN,True,NaN,NaN,NaN,199000.0,529618166,carolthefinder@yahoo.com,2024-01-08,141500.0,CAROL,SMITH,34.396733,-117.220954,20171 Riverview Road,Land,NaN,145000.0,830,Keller Williams Realty Phelan,Coldwell Banker Home Source,Keller Williams Realty Phelan,CAROL SMITH,DINA,VELEZ,HD12177,Stacey,Hart,NaN,NaN,NaN,529618166,NaN,NaN,San Bernardino,Closed,NaN,NaN,NaN,NaN,NaN,5.0000,NaN,HighDesert,NaN,Item1,20171.0,537165,NaN,Apple Valley,3.0,NaN,NaN,NaN,2024-01-08,NaN,NaN,2023-10-24,2021-07-14,NaN,NaN,CA,NaN,NaN,NaN,NaN,NaN,NaN,Irregular,217800.0,NaN,False,NaN,NaN,92308,NaN,217800.0,NaN,NaN,NaN,NaN,NaN
4,NaN,True,NaN,NaN,NaN,19500.0,522614340,jtavisola@tavisola.com,2024-01-17,15000.0,Jeremie,Tavisola,34.454756,-117.782248,0 Vac/Vic Avenue Z/195 Ste,Land,NaN,19500.0,805,Realty ONE Group West,Realty ONE Group West,NaN,Jeremie Tavisola,NaN,NaN,TTaviJer,Jeremie,Tavisola,NaN,NaN,NaN,522614340,LLO - Llano,NaN,Los Angeles,Closed,NaN,NaN,NaN,NaN,NaN,2.4996,NaN,S

In [ ]:
sold.dtypes

#### Unique property types found

In [367]:
sold["PropertyType"].value_counts()

PropertyType
Residential            458336
ResidentialLease       155507
Land                    22601
ManufacturedInPark      18604
ResidentialIncome       18212
CommercialSale           4267
CommercialLease          3599
BusinessOpportunity       473
Name: count, dtype: int64

#### Focusing on residential properties, including:
* "Residential": standard homes intended for owner-occupany
* "ResidentialLease": residential properties offered for rent
* "ResidentialIncome": multi-unit residential properties purchased primarily for rental income

I chose to focus on traditional residential home sales and excluded "ManufacturedInPark" homes (mobile homes) located in mobile home parks because they differ from site-built in terms of ownership structure, financing, depreciation, and market dyanmics. Since  mix mobil/manufactured homes and traditional site-built residential represent fundamentally different asset classes analyzing them together can severely skew metrics.

After filtering, the new dataset for sales of traditional, residential homes decreases from 681599 to 632055.

In [368]:
sold_resid = sold[sold["PropertyType"].isin(["Residential", "ResidentialLease", "ResidentialIncome"])]

#### Validate completeness for sold data

In [ ]:
null_report_sold = pd.DataFrame({
    "Null Count": sold_resid.isnull().sum(),
    "Null %": (sold_resid.isnull().sum() / len(sold_resid) * 100).round(2)
})

null_report_sold["Flag (>90% null)"] = null_report_sold["Null %"] > 90

null_report_sold = null_report_sold.sort_values("Null %", ascending=False)
null_report_sold

I dropped columns with 90% or more missing values to improve the quality and reliability of my dataset. These columns did not contain enough data to support meaningful analysis, and any results based on them would likely be unstable or biased due to the very small number of valid entries. Removing them also helped simplify my dataset and reduce noise, making my analysis more accurate and easier to interpret.

I also dropped 6 rows with missing closing price because the closing price is essential for analyzing sales and calculating meaningful metrics like average price, trends, and comparisons. Without it, those records cannot be reliably used in price-based analysis, so removing them ensures the results are accurate and consistent.

After dropping, there are now 632049 observations in the sold_residential dataset

In [370]:
drop_cols_sold = null_report_sold[null_report_sold["Flag (>90% null)"]].index
sold_resid = sold_resid.drop(columns=drop_cols_sold)

sold_resid = sold_resid.dropna(subset=["ClosePrice"])

#### Numeric distribution summary, Sold (min, max, mean, median, percentiles) for ClosePrice, LivingArea, and DaysOnMarket

In [371]:
cols = ['ClosePrice', 'LivingArea', 'DaysOnMarket']

summary_sold = pd.DataFrame({
    'min': sold_resid[cols].min(),
    'max': sold_resid[cols].max(),
    'mean': sold_resid[cols].mean(),
    'median': sold_resid[cols].median(),
    '25%': sold_resid[cols].quantile(0.25),
    '75%': sold_resid[cols].quantile(0.75)
})

summary_sold

,min,max,mean,median,25%,75%
ClosePrice,0.0,989500000.0,909605.963810,665000.0,165000.0,1104000.0
LivingArea,0.0,909090909.0,3326.839201,1596.0,1188.0,2173.0
DaysOnMarket,-288.0,12430.0,38.138878,20.0,8.0,49.0


#### Repeat for listing dataset

In [ ]:
list.columns

In [373]:
list.head()

,OriginalListPrice,ListingKey,ListAgentEmail,CloseDate,ClosePrice,ListAgentFirstName,ListAgentLastName,Latitude,Longitude,UnparsedAddress,PropertyType,LivingArea,ListPrice,DaysOnMarket,ListOfficeName,BuyerOfficeName,CoListOfficeName,ListAgentFullName,CoListAgentFirstName,CoListAgentLastName,BuyerAgentMlsId,BuyerAgentFirstName,BuyerAgentLastName,FireplacesTotal,AssociationFeeFrequency,AboveGradeFinishedArea,ListingKeyNumeric,MLSAreaMajor,TaxAnnualAmount,CountyOrParish,PropertyType.1,MlsStatus,ElementarySchool,ListAgentFirstName.1,AttachedGarageYN,ParkingTotal,BuilderName,PropertySubType,LotSizeAcres,SubdivisionName,BuyerOfficeAOR,YearBuilt,DaysOnMarket.1,BuyerAgencyCompensationType,StreetNumberNumeric,LivingArea.1,ListingId,BathroomsTotalInteger,City,BuyerAgencyCompensation,TaxYear,BuildingAreaTotal,BedroomsTotal,ContractStatusChangeDate,Longitude.1,ElementarySchoolDistrict,CoBuyerAgentFirstName,PurchaseContractDate,ListingContractDate,BelowGradeFinishedArea,BusinessType,Latitude.1,ListPrice.1,StateOrProvince,CoveredSpaces,MiddleOrJuniorSchool,FireplaceYN,Stories,HighSchool,Levels,ListAgentLastName.1,CloseDate.1,LotSizeDimensions,LotSizeArea,MainLevelBedrooms,NewConstructionYN,GarageSpaces,HighSchoolDistrict,PostalCode,BuyerOfficeName.1,AssociationFee,LotSizeSquareFeet,MiddleOrJuniorSchoolDistrict,UnparsedAddress.1
0,90000.0,1075010398,miriamlara03@gmail.com,NaN,NaN,Miriam,Lara,34.097939,-117.909653,1045 N Azusa 61,ManufacturedInPark,960.0,90000.0,0,CENTURY 21 CITRUS REALTY INC,NaN,NaN,Miriam Lara,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,1075010398,614 - Covina,NaN,Los Angeles,ManufacturedInPark,Active,NaN,Miriam,False,4.0,NaN,NaN,NaN,NaN,NaN,1960.0,0,Item,1045.0,960.0,CV24093347,1.0,Covina,2500.0,NaN,NaN,4.0,2024-01-29,-117.909653,NaN,NaN,NaN,2024-01-29,NaN,NaN,34.097939,90000.0,CA,NaN,NaN,NaN,1.0,NaN,One,Lara,NaN,NaN,NaN,NaN,NaN,2.0,Covina Valley Unified,91722,NaN,0.0,NaN,NaN,1045 N Azusa 61
1,1500000.0,1074974457,janelle@judsonre.com,NaN,NaN,Janelle,Judson,33.121241,-117.081614,NaN,CommercialSale,NaN,1500000.0,0,Judson Real Estate,NaN,Judson Real Estate,Janelle Judson,Lynn,Judson,NaN,NaN,NaN,NaN,NaN,NaN,1074974457,92025 - Escondido,NaN,San Diego,CommercialSale,Active,NaN,Janelle,NaN,0.0,NaN,Retail,NaN,NaN,NaN,1940.0,0,Item1,NaN,NaN,NDP2403932,0.0,Escondido,2.0,NaN,3000.0,NaN,2024-01-11,-117.081614,NaN,NaN,NaN,2024-01-11,NaN,NaN,33.121241,1500000.0,CA,NaN,NaN,NaN,NaN,NaN,NaN,Judson,NaN,NaN,3052.0,NaN,NaN,NaN,NaN,92025,NaN,0.0,NaN,NaN,NaN
2,1340000.0,1074973329,haleh360@Gmail.com,NaN,NaN,Haleh,Dowlatshahi,34.052207,-118.408445,2220 Avenue Of The Stars 2704,Residential,1301.0,1340000.0,127,Rodeo Realty- Brentwood,NaN,NaN,Haleh Dowlatshahi,NaN,NaN,NaN,NaN,NaN,NaN,Monthly,NaN,1074973329,C05 - Westwood - Century City,NaN,Los Angeles,Residential,Pending,NaN,Haleh,NaN,0.0,NaN,Condominium,4.0831,NaN,NaN,1964.0,127,Item1,2220.0,1301.0,24389291,2.0,Los Angeles,2.0,NaN,1301.0,2.0,2024-05-07,-118.408445,NaN,NaN,2024-05-07,2024-01-01,NaN,NaN,34.052207,1340000.0,CA,NaN,NaN,False,1.0,NaN,One,Dowlatshahi,NaN,NaN,177861.0,NaN,False,NaN,NaN,90067,NaN,2105.0,177861.0,NaN,2220 Avenue Of The Stars 2704
3,2500000.0,1074954552,Reneechen@yourhomesoldguaranteed.com,NaN,NaN,Renee,Chen,33.496363,-117.691677,16 Palisades,Residential,2788.0,2500000.0,1,Your Home Sold Guaranteed Realty,NaN,NaN,Renee Chen,NaN,NaN,NaN,NaN,NaN,NaN,Monthly,NaN,1074954552,LNSLT - Salt Creek,NaN,Orange,Residential,Active,NaN,Renee,True,3.0,NaN,SingleFamilyResidence,0.1217,Warmington Homes (BB) (BBW),NaN,1994.0,1,Item1,16.0,2788.0,AR24091628,3.0,Laguna Niguel,2.0,NaN,NaN,4.0,2024-01-24,-117.691677,NaN,NaN,NaN,2024-01-24,NaN,NaN,33.496363,2500000.0,CA,NaN,NaN,True,2.0,NaN,Two,Chen,NaN,NaN,5300.0,0.0,False,3.0,Capistrano Unified,92677,NaN,254.0,5300.0,NaN,16 Palisades
4,3150000.0,1074936537,anader@dppre.com,NaN,NaN,Margaret,Nader,34.119345,-118.111254,1615 Waverly Road,Residential,3250.0,3150000.0,1,COMPASS,NaN,NaN,Margaret Nader,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,1074936537,655 - San Marino,

In [ ]:
list.dtypes

In [375]:
list["PropertyType"].value_counts()

PropertyType
Residential            567549
ResidentialLease       185595
Land                    58501
ResidentialIncome       33187
ManufacturedInPark      25641
CommercialSale          12198
CommercialLease          8070
BusinessOpportunity      2853
Name: count, dtype: int64

#### Applying same residential classification and filter to listing data

In [376]:
list_resid = list[list["PropertyType"].isin(["Residential", "ResidentialLease", "ResidentialIncome"])]

#### Validate completeness for listing data

In [ ]:
null_report_list = pd.DataFrame({
    "Null Count": list_resid.isnull().sum(),
    "Null %": (list_resid.isnull().sum() / len(list_resid) * 100).round(2)
})

null_report_list["Flag (>90% null)"] = null_report_list["Null %"] > 90

null_report_list = null_report_list.sort_values("Null %", ascending=False)
null_report_list

Applying the same logic for the sold data, I also dropped columns with 90% or more missing values for the listing data.

I did not drop observations with missing closing price as it is natural and justified that not all real estate listed is going to be sold within the time frame the data was pulled.

In [378]:
drop_cols_list = null_report_list[null_report_list["Flag (>90% null)"]].index
list_resid = list_resid.drop(columns=drop_cols_list)

#### Numeric distribution summary, Listing (min, max, mean, median, percentiles) for ClosePrice, LivingArea, and DaysOnMarket

In [379]:
cols = ['ClosePrice', 'LivingArea', 'DaysOnMarket']

summary_list = pd.DataFrame({
    'min': list_resid[cols].min(),
    'max': list_resid[cols].max(),
    'mean': list_resid[cols].mean(),
    'median': list_resid[cols].median(),
    '25%': list_resid[cols].quantile(0.25),
    '75%': list_resid[cols].quantile(0.75)
})

summary_list

,min,max,mean,median,25%,75%
ClosePrice,0.0,820000000.0,797204.109929,590000.0,4800.0,1050000.0
LivingArea,0.0,909090909.0,3159.752698,1624.0,1192.0,2266.0
DaysOnMarket,-58.0,2455.0,18.432450,10.0,5.0,21.0


#### Save the filtered dataset as a new CSV (for different types of residential property types and drops)

In [380]:
list_resid.to_csv("w2_list_resid.csv", index=False)
sold_resid.to_csv("w2_sold_resid.csv", index=False)

### **Week 3** — Mortgage Rate Enrichment